libs/ledger-live-common/src/families/cosmos/api/Cosmos.ts

```diff
@@ -229,49 +229,94 @@ export class CosmosAPI {
     return data.withdraw_address;
   };
 
-  getTransactions = async (address: string): Promise<CosmosTx[]> => {
-    let receive;
-    let send;
+  getTransactions = async (address: string, paginationSize: number): Promise<CosmosTx[]> => {
+    const senderTxs = await this.fetchAllTransactions(address, "message.sender", paginationSize);
+    const recipientTxs = await this.fetchAllTransactions(
+      address,
+      "transfer.recipient",
+      paginationSize,
+    );
 
+    return [...senderTxs, ...recipientTxs];
+  };
+
+  private async fetchAllTransactions(address: string, filterOn: string, paginationSize: number) {
```

Comment:
If we know what is parameter on `filterOn` shouldn't we just specify it with : `"message.sender" || "transfert.recipient"`
Being less permissive sounds better to me

Reply:
In our app we use those two values but any filter on a transaction could be used, I don't know what is the best way to do it :shrug: 

I can do as you said, restrict on what we know is used on the app

Reply:
If we can be more precise it's better for typing

Reply:
Done